### DISCUSSION TITLE
Problemi di aggiornamento Home Assistant Core su Raspberry Pi OS

### DISCUSSION THREAD

**Discussion by alfrefly** [indomus/forum#1548] [HUB: Home Assistant]
Salve, ho un'installazione perfettamente funzionante di HA Core su Raspberry Pi OS, immagine Phoscon.
 La versione di HA è la 2021.11.1, Python 3.8.6 e Pip 20.3.4
Per poter aggiornare la nuova versione di HA devo aggiornare Python ad una versione più nuova, seguendo questa guida:

https://indomus.it/guide/installare-python-3-9-x-o-superiori-su-raspberry-pi-os-e-ripristinare-home-assistant-core-su-venv/

dopo l'installazione di python e l'aggiornamento di HA, non funziona più.
Ho anche provato a fare varie installazioni su una nuova SD prima ho installato l'ultima versione di HA e poi installato Python 3.10.2 e riavviato HA, ma non funziona. La seconda ho aagirnato Python prima di installarci sopra HA ma non riesco a portare a termine l'installazione di HA, l'errore è il seguente:

  This package requires Rust >=1.41.0.
  ----------------------------------------
  ERROR: Failed building wheel for cryptography
  Building wheel for pyyaml (PEP 517) ... done
  Created wheel for pyyaml: filename=PyYAML-6.0-cp310-cp310-linux_armv7l.whl size=45332 sha256=568d547861a93538c4ac0fad912654d9f903d6142ff61c9fd92582bb36d37ab9

Cosa sbaglio?

**Comment by snakuzzo** [comment]
Leggi [QUI](https://indomus.it/guide/come-installare-e-configurare-home-assistant-su-un-raspberry-pi-gia-in-uso/#rust)

**Reply by Scialla** [reply]
Inoltre HA non è ancora completamente compatibile con python 3.10, ci va il 3.9.x

Valuta il passaggio a installazione con docker, tutti sti problemi non li hai più..
